app/src/main/java/com/cornellappdev/transit/models/RouteRepository.kt

```
@@ -142,15 +146,30 @@ class RouteRepository @Inject constructor(
         CoroutineScope(Dispatchers.IO).launch {
             try {
                 val placeResponse = appleSearch(SearchQuery(query))
                 val res = placeResponse.unwrap()
                 val totalLocations = (res.places ?: emptyList()) + (res.stops ?: (emptyList()))
-                _placeFlow.value = ApiResponse.Success(totalLocations)
+                if (token == latestSearchToken.get()) {
+                    _placeFlow.value = ApiResponse.Success(totalLocations)
+                }
+            } catch (_: kotlinx.coroutines.CancellationException) {
+                // Ignore cancellation; latest query owns the flow update.
             } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)